infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -237,6 +265,12 @@ public LogReplicationEntryMsg resend() {
             }
         } catch (Exception e) {
             log.warn("Caught an exception while processing ACKs.", e);
+        } finally {
+            if (isLogEntry && force) {
+                activateBackpressure();
+            } else {
+                deactivateBackpressure();
```

Comment:
We would come here even if its a snapshot sync...can we restructure this block so we have a top level if statement checking for isLogEntry, and a child if-else block activating/deactivating the back pressure?

Reply:
That behavior was intentional in the event that log entry was interrupted by a force sync then that snapshot sync would reset the back pressure when it calls resend, but deactivating when we transition out out log entry and making this explicit is a better approach so will take that up.